### DISCUSSION TITLE
Idea's for financial math algorithms

### DISCUSSION THREAD

**Discussion by fareswmansi** [TheAlgorithms/Python#8922] [Ideas]
Hey everyone!

I have some ideas for algorithms to implement for financial math in this repository. I would like to implement some of these and hope others do the same.

To add to simple interest:
- algorithm to convert between discount rate to interest rate

 Immunization :
- Algorithm to calculate Macaulay duration and convexity
- algorithm to calculate modified duration and convexity
- algorithm to dictate Full or Redington immunization (utilizes Macaulay duration and modified duration algorithms)

Links:
Macaulay duration: https://www.investopedia.com/terms/m/macaulayduration.asp
Modified duration: https://www.investopedia.com/terms/m/modifiedduration.asp
Immunization: https://en.wikipedia.org/wiki/Immunization_(finance)

**Comment by L-Huber** [comment]
Sounds good. I would love to see more finance algorithms.
